### PR TITLE
[Buttons] Deprecate subclassing to set cornerRadius and contentEdgeInsets

### PR DESCRIPTION
**Pull request by samnm**
Previously cornerRadius and contentEdgeInsets were only configurable via subclassing MDCButton. This PR deprecates that behavior in favor of setting the properties explicitly.

### CODE REVIEW

File: components/Buttons/src/MDCButton.m [hunk 1]
```diff
@@ -51,6 +51,9 @@
 
 static const NSTimeInterval MDCButtonAnimationDuration = 0.2;
 
+static const CGFloat MDCButtonDefaultCornerRadius = 2.0;
+const UIEdgeInsets MDCButtonDefaultContentEdgeInsets = {8, 16, 8, 16};
```
Comment (ianegordon) [inline components/Buttons/src/MDCButton.m:64]: These are 0,0,0,0 on standard UIButtons.
Add a comment / documentation on why we are different.

File: components/Buttons/src/MDCRaisedButton.m [hunk 2]
```diff
@@ -26,6 +25,7 @@ + (void)initialize {
                                     forState:UIControlStateNormal];
   [[MDCRaisedButton appearance] setElevation:MDCShadowElevationRaisedButtonPressed
                                     forState:UIControlStateHighlighted];
+  [[MDCRaisedButton appearance] setContentEdgeInsets:MDCButtonDefaultContentEdgeInsets];
```
Comment (ianegordon) [inline components/Buttons/src/MDCRaisedButton.m:28]: Isn't this already set in +[MDCButton initialize] ?

File: components/Buttons/src/MDCFlatButton.m [hunk 3]
```diff
@@ -33,6 +33,7 @@ + (void)initialize {
                                   forState:UIControlStateNormal];
   [[MDCFlatButton appearance] setElevation:MDCShadowElevationNone
                                   forState:UIControlStateHighlighted];
+  [[MDCFlatButton appearance] setContentEdgeInsets:MDCButtonDefaultContentEdgeInsets];
```
Comment (ianegordon) [inline components/Buttons/src/MDCFlatButton.m:36]: Already done in +[MDCButton initialize].  Do we need to do it again?

**Review by ianegordon — CHANGES_REQUESTED** [review]

File: components/Buttons/src/MDCRaisedButton.m (hunk 2 above)
Comment (samnm) [reply components/Buttons/src/MDCRaisedButton.m:28]: UIAppearance calls applied to MDCButton don't apply to subclasses of MDCButton.

File: components/Buttons/src/MDCButton.m (hunk 1 above)
Comment (samnm) [reply components/Buttons/src/MDCButton.m:64]: Added a comment.

File: components/Buttons/src/MDCRaisedButton.m (hunk 2 above)
Comment (ianegordon) [reply components/Buttons/src/MDCRaisedButton.m:28]: Of course. ack.

File: components/Buttons/src/MDCFlatButton.m (hunk 3 above)
Comment (ianegordon) [reply components/Buttons/src/MDCFlatButton.m:36]: Of course. ack.

**Review by ianegordon — APPROVED** [review]

**Comment by ianegordon** [comment]
This PR is not intended to change any functionality, but to change the interface that exposes the functionality.

**Comment by ianegordon** [comment]
Please merge when Travis is happy.

**Pull request merged**